src/src/PluginStructs/P012_data_struct.cpp

```diff
@@ -96,11 +96,10 @@ void P012_data_struct::lcdWrite(const String& text, uint8_t col, uint8_t row) {
     }
   }
 
+  if (row == 0) { firstLineState = 0; } // Reset firstLineState
```

Comment:
Shouldn't you also clear the entire line, or else some left-over from the splashscreen may still be visible.

Reply:
That should be taken care of by the `LCD command mode` setting. The actual clearing is only done when the timer expires, by sending exactly 8 spaces, to avoid clearing something that arrives between the last check and the timer expiring (though it's _close_ to impossible to interfere with that).

Reply:
But what if you call from the rules to start the task and write a line to it?
The flag is then cleared, so a call to PLUGIN_READ should not clear the line anymore.
And maybe a call initiated from the PLUGIN_INIT should not call for a PLUGIN_READ, but for a `PLUGIN_TASKTIMER_IN` ???

Reply:
When writing something on line 1 (0 internally) via rules passes through that `lcdWrite()` method so it will reset the flag, and clearing is ignored on the next PLUGIN_READ.
I could (should?) have used an enum with descriptive names here, not sure if that would add to the code-size, as I tried to keep it as lean as possible, this plugin is available in most builds...

Reply:
You can always use defines to clarify 'magic' values.

Reply:
Using an enum class of uint8_t proved to be the most compact on ESP8266 😉